test/tech/v3/ml_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns tech.v3.ml-test
+  (:require [tech.v3.dataset :as ds]
+            [tech.v3.dataset.modelling :as ds-mod]
+            [tech.v3.libs.smile.discrete-nb :as nb]
+            [tech.v3.libs.smile.nlp :as nlp]
+            [tech.v3.ml :as ml]
+            [tech.v3.ml.gridsearch :as ml-gs]))
+
+(comment
+
+  (def dataset
+    (->
+     (ds/->dataset "test/data/reviews.csv.gz" {:key-fn keyword })
+     (ds/select-columns [:Text :Score])
+     (ds/update-column :Score #(map dec %))
+     (nlp/count-vectorize :Text :bow nlp/default-text->bow)
+     (ds-mod/set-inference-target :Score)))
+
+
+  (defn preprocess [ds options]
```

Comment:
This is the pipeline specific preprocess function, adn it want to grid search over parameter "vocab-size"